### DISCUSSION TITLE
Can I use AlphaPullDown for 5K proteins?

### DISCUSSION THREAD

**Discussion by Rohit-Satyam** [KosinskiLab/AlphaPulldown#430] [Q&A]
I was wondering if AlphaPulldown will complain if given 5000 proteins to perform all-vs-all PPI. Pardon my ignorance but in the Document I couldn't find a place where we can skip running structure prediction and use the ones made available by Alphafold database to reduce the runtime.

**Comment by jkosinski** [comment]
Hi Rohit-Satyam,

AP will not complain but it will take a lot of time or resources. We thought of using monomeric models for speeding up the calculations but we haven’t tested that, in principle should work by using the models as custom monomeric templates and reducing the number of recycles. @DimaMolod , would using custom monomeric templates work for all against all mode?

**Comment by DimaMolod** [comment]
Hi, yes, using monomeric templates would work, but it will not speed up predictions as searching for a template is rapid compared to the rest of the workflow. Reducing cycles to 1 will help, but the quality of predictions will (expectedly) deteriorate. Please also note that 5000 in all-vs-all will result in 12.5*10^6 predictions, which is too much even for the most powerful HPCs in the world

**Comment by jkosinski** [comment]
There is a chance that the quality of models would still be acceptable with recycle 1 as the monomeric templates may speed up "convergence". But indeed, we haven't tested that yet; it's just a speculation.

**Comment by jkosinski** [comment]
> Hi, yes, using monomeric templates would work, but it will not speed up predictions as searching for a template is rapid compared to the rest of the workflow. Reducing cycles to 1 will help, but the quality of predictions will (expectedly) deteriorate. Please also note that 5000 in all-vs-all will result in 12.5*10^6 predictions, which is too much even for the most powerful HPCs in the world

Doable in a week on this just everyone else at xAI needs to go for holidays https://www.reddit.com/r/artificial/comments/1f8iidw/musks_xai_supercomputer_goes_online_with_100000/

**Comment by Rohit-Satyam** [comment]
I get it that it's quite ambitious. We don't have that many GPUs (600 available but shared) but we have 1000 CPUs. Will somehow reduce the list of proteins for all vs all comparison.

**Comment by jkosinski** [comment]
Dear Rohit,

CPUs will not help you much unfortunately. Indeed best to filter the list somehow. You could also try to prefilter using the approach from https://www.biorxiv.org/content/10.1101/2024.10.01.615885v1 but I don’t know how long that would take for all pairs of 5000 proteins and you may miss interactions.

Best,
Jan
